frontend/src/pages/admin/portal-edit.js

```diff
@@ -0,0 +1,483 @@
+    delete portal.owner;
+    // delete portal.defaultDashboard;
+    this.savePortal(portal.id || null, portal)
+      .then(() => this.props.navigate(-1))
+      .catch((error) => console.error(error));
+  };
+
+  onOwnerToggle = () => {
+    this.setState({isOwnerOpen: !this.state.isOwnerOpen});
+  };
+
+  onDashboardToggle = () => {
+    this.setState({isDashboardOpen: !this.state.isDashboardOpen});
+  };
+
+  onOwnerInputChange = (_event, value) => {
+    this.setState({inputValueOwner: value});
+    this.setState({filterValueOwner: value});
+  };
+
+  onOwnerSelect = (event, value) => {
+    this.setState({
+      selectedOwner: value.user,
+      isOwnerOpen: false,
+      filterValueOwner: '',
+      inputValueOwner: value.user.name
+    });
+  };
+
+  onOwnerClear = () => {
+    this.setState({
+      selectedOwner: null,
+      inputValueOwner: '',
+      filterValueOwner: ''
+    });
+  }
+
+  getPortal(portalId) {
+    HttpClient.get([Settings.serverUrl, 'admin', 'portal', portalId])
+      .then(response => {
+        response = HttpClient.handleResponse(response, 'response');
+        return response.json();
+      })
+      .then(portal => {
+        this.setState({portal: portal, selectedOwner: portal.owner,
+                       inputValueOwner: portal.owner?.name,
+                       selectedDashboard: portal.defaultDashboard,
+                       inputValueDashboard: portal.defaultDashboard?.title});
+      })
+      .catch(error => console.error(error));
+  }
+
+  onDashboardToggle = () => {
+    this.setState({isDashboardOpen: !this.state.isDashboardOpen});
+  };
+
+  onDashboardSelect = (event, value) => {
+    this.setState({
+      selectedDashboard: value.dashboard,
+      isDashboardOpen: false,
+      filterValueDashboard: '',
+      inputValueDashboard: value.dashboard.title
+    });
+  };
+
+  onDashboardClear = () => {
+    this.setState({
+      selectedDashboard: null,
+      inputValueDashboard: '',
+      filterValueDashboard: ''
+    });
+  }
+
+  onDashboardInputChange = (_event, value) => {
+    this.setState({inputValueDashboard: value});
+    this.setState({filterValueDashboard: value});
+  };
+
+  getDashboards() {
+    if (!this.state.id || this.state.id == "new" || !this.state.portal) {
+      return;
+    }
+    let params = {
+      'portal_id': this.state.id,
+      'pageSize': 10
+    };
+    HttpClient.get([Settings.serverUrl, 'dashboard'], params)
+      .then(response => HttpClient.handleResponse(response))
+      .then(data => this.setState({dashboards: data['dashboards'], filteredDashboards: data['dashboards']}));
+  }
+
+  getUsers() {
+    HttpClient.get([Settings.serverUrl, 'admin', 'user'])
+      .then(response => {
+        response = HttpClient.handleResponse(response, 'response');
+        return response.json();
+      })
+      .then(data => this.setState({users: data.users, filteredUsers: data.users}))
+      .catch(error => console.error(error));
```

Comment:
Console statement

Reply:
ack - these errors are intentionally console logged right now (at least they are in the project components).

I'll look at better handling them so we have them in pod logs too?